### PR TITLE
fix: Vim Chord and Vim Mode display freeze / stop working when switch…

### PR DESCRIPTION
**Pull request by jiyee**
- FIX: Vim Chord and Vim Mode display freeze / stop working when switched to a new note, on another tab and after some time #149 
- Support source command for using .vimrc in home directory #157

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -509,31 +547,37 @@ export default class VimrcPlugin extends Plugin {
...
+			tempS += " " + s;
+		}
+		if (tempS != "") {
+			tempS += "-";
 		}
+		this.vimChordStatusBar.setText(tempS);
+	}
+
+	onVimCommandDone = async (reason: any) => {
+		this.vimChordStatusBar.setText("");
```
Comment (esm7) [inline main.ts:579]: This throws an exception if chord display is off

File: main.ts [hunk 2]
```diff
@@ -543,18 +587,16 @@ export default class VimrcPlugin extends Plugin {
...
-		cm.on('keydown', (instance: CodeMirror.Editor, ev: KeyboardEvent) => {
-			if (this.settings.fixedNormalModeLayout) {
-				const keyMap = this.settings.capturedKeyboardMap;
-				if (!this.isInsertMode && !ev.shiftKey &&
-					ev.code in keyMap && ev.key != keyMap[ev.code]) {
-					this.codeMirrorVimObject.handleKey(instance, keyMap[ev.code], 'mapping');
-					ev.preventDefault();
-					return false;
-				}
+	onKeydown = (instance: CodeMirror.Editor, ev: KeyboardEvent) => {
```
Comment (esm7) [inline main.ts:590]: I may be missing something but this doesn't seem to work for me. A log put before the `if` never seems to be printed. Any clue why?

File: main.ts (hunk 1 above)
Comment (jiyee) [reply main.ts]: fixed.

File: main.ts (hunk 2 above)
Comment (jiyee) [reply main.ts]: It seems like the way of subscribe CodeMirror event diffs, I try to fix it to work, but I don't know how to debugger in Normal Mode.

**Comment by esm7** [comment]
Looking good, thank you for your work on this!

**Comment by huyz** [comment]
Weird, my bindings no longer work and I had to downgrade to v0.8.0:
```vim
exmap TableFormat obcommand table-editor-obsidian:format-table
map µf :TableFormat
```

It seems the code no longer likes the special character `µ` which I generate in macOS by typing <kbd>⌥m</kbd>

**Pull request merged**